### PR TITLE
rustfmt :- rustfmt the project

### PR DESCRIPTION
**Pull request by naveensrinivasan**
- rustfmt the project
- included additional targets for the make file

## Description

Fixes # (issue)

 

### Type of change

- [ ] Bug fix (non-breaking change which fixes an issue)
- [ ] New feature (non-breaking change which adds functionality)
- [ ] Breaking change (fix or feature that would cause existing functionality to not work as expected)
- [ ] This change requires a documentation update

## How Has This Been Tested?

- Test A - `lind_project/tests/test_cases/test_a.c`
- Test B - `lind_project/tests/test_cases/test_b.c`

## Checklist:

- [ ] My code follows the style guidelines of this project
- [ ] I have commented my code, particularly in hard-to-understand areas
- [ ] My changes generate no new warnings
- [ ] I have added tests that prove my fix is effective or that my feature works
- [ ] Any dependent changes have been added to a pull request and/or merged in other modules (native-client, lind-glibc, lind-project)

### CODE REVIEW

File: benches/fs_open_close.rs [hunk 1]
```diff
@@ -20,62 +17,56 @@ use rustposix::safeposix::cage::*;
...
-            }
-        }
-    ));
+    group.bench_function("TF01: Native OS kernel open+close", |b| {
+        b.iter(|| unsafe {
+            let fd = libc::open(
+                CString::new("/tmp/foo").unwrap().as_ptr(),
+                O_CREAT | O_TRUNC | O_WRONLY,
+                S_IRWXA,
+            );
```
Comment (JustinCappos) [inline benches/fs_open_close.rs:57]: this change here is a little weird.  It added a trailing "," on the last arg...

File: src/safeposix/net.rs [hunk 2]
```diff
@@ -19,31 +19,43 @@ pub const TCPPORT: bool = true;
...
+            )),
+            next_ephemeral_port_udpv4: interface::RustRfc::new(interface::RustLock::new(
+                EPHEMERAL_PORT_RANGE_END,
+            )),
+            next_ephemeral_port_tcpv6: interface::RustRfc::new(interface::RustLock::new(
+                EPHEMERAL_PORT_RANGE_END,
+            )),
+            next_ephemeral_port_udpv6: interface::RustRfc::new(interface::RustLock::new(
+                EPHEMERAL_PORT_RANGE_END,
+            )),
```
Comment (JustinCappos) [inline src/safeposix/net.rs:36]: These are a bit silly, but not to the point of fighting the formatter...

File: src/safeposix/syscalls/net_constants.rs [hunk 3]
```diff
@@ -131,149 +131,147 @@ pub const PF_MAX: i32 = AF_MAX;
...
+pub const IPPROTO_SWIPE: i32 = 53; // IP with encryption
+pub const IPPROTO_NHRP: i32 = 54; // Next Hop Resolution
+                                  // 55-57: Unassigned
+pub const IPPROTO_ICMPV6: i32 = 58; // ICMP6
+pub const IPPROTO_NONE: i32 = 59; // IP6 no next header
+pub const IPPROTO_DSTOPTS: i32 = 60; // IP6 destination option
+pub const IPPROTO_AHIP: i32 = 61; // any host internal protocol
+pub const IPPROTO_CFTP: i32 = 62; // CFTP
+pub const IPPROTO_HELLO: i32 = 63; // "hello" routing protocol
+pub const IPPROTO_SATEXPAK: i32 = 64; // SATNET/Backroom EXPAK
```
Comment (JustinCappos) [inline src/safeposix/syscalls/net_constants.rs:196]: Let's exempt this file from rust_fmt...

File: src/safeposix/syscalls/sys_calls.rs [hunk 4]
```diff
@@ -127,57 +159,87 @@ impl Cage {
...
+        if !interface::RUSTPOSIX_TESTSUITE.load(interface::RustAtomicOrdering::Relaxed) {
+            // we don't add these for the test suite
+            let mainsigsetatomic = self
+                .sigset
+                .get(
+                    &self
+                        .main_threadid
+                        .load(interface::RustAtomicOrdering::Relaxed),
+                )
+                .unwrap();
```
Comment (JustinCappos) [inline src/safeposix/syscalls/sys_calls.rs:204]: a bit fiddly here...

File: src/safeposix/syscalls/sys_constants.rs [hunk 5]
```diff
@@ -9,71 +9,69 @@ use crate::interface;
...
 
-pub const SIG_BLOCK: i32    = 0;
-pub const SIG_UNBLOCK: i32  = 1;
-pub const SIG_SETMASK: i32  = 2;
-pub const ITIMER_REAL: i32  = 0;
\ No newline at end of file
+pub const SIG_BLOCK: i32 = 0;
+pub const SIG_UNBLOCK: i32 = 1;
+pub const SIG_SETMASK: i32 = 2;
+pub const ITIMER_REAL: i32 = 0;
```
Comment (JustinCappos) [inline src/safeposix/syscalls/sys_constants.rs:77]: should be exempted

File: src/tests/fs_tests.rs [hunk 6]
```diff
@@ -125,37 +119,52 @@ pub mod fs_tests {
...
-        assert_eq!(cage.pwrite_syscall(fd, str2cbuf("Lorem ipsum dolor sit amet, consectetur adipiscing elit"), 55, 0), 55);
+        assert_eq!(
+            cage.pwrite_syscall(
+                fd,
+                str2cbuf("Lorem ipsum dolor sit amet, consectetur adipiscing elit"),
+                55,
+                0
+            ),
+            55
+        );
```
Comment (JustinCappos) [inline src/tests/fs_tests.rs:137]: ugly.  Original was better...

File: src/tests/fs_tests.rs [hunk 7]
```diff
@@ -125,37 +119,52 @@ pub mod fs_tests {
...
+            cage.pread_syscall(fd, read_bufzero.as_mut_ptr(), 1000, 0),
+            1000
+        );
+        assert_eq!(
+            cbuf2str(&read_bufzero),
+            std::iter::repeat("\0")
+                .take(1000)
+                .collect::<String>()
+                .as_str()
+        );
```
Comment (JustinCappos) [inline src/tests/fs_tests.rs:150]: I'm not sure we want the change here either...

File: src/tests/ipc_tests.rs [hunk 8]
```diff
@@ -107,72 +109,149 @@ pub mod ipc_tests {
...
+            assert_eq!(
+                cage2.recvfrom_syscall(
+                    sockfd,
+                    buf.as_mut_ptr(),
+                    100,
+                    MSG_PEEK,
+                    &mut Some(&mut socket2)
+                ),
+                100
+            ); //peeking at the input message
```
Comment (JustinCappos) [inline src/tests/ipc_tests.rs:135]: doesn't really seem clearer to me...

File: src/tests/ipc_tests.rs [hunk 9]
```diff
@@ -107,72 +109,149 @@ pub mod ipc_tests {
...
+            assert_eq!(
+                cage2.recvfrom_syscall(
+                    sockfd,
+                    buf.as_mut_ptr(),
+                    20,
+                    MSG_PEEK,
+                    &mut Some(&mut socket2)
+                ),
+                20
+            );
```
Comment (JustinCappos) [inline src/tests/ipc_tests.rs:164]: likewise, not really clearer

File: src/tests/mod.rs [hunk 10]
```diff
@@ -7,31 +7,58 @@ mod networking_tests;
...
+                0
+            );
+            assert_eq!(
+                cage.mknod_syscall(
+                    "/dev/random",
+                    S_IFCHR as u32 | 0o777,
+                    makedev(&DevNo { major: 1, minor: 8 })
+                ),
+                0
+            );
```
Comment (JustinCappos) [inline src/tests/mod.rs:61]: seems a little too aggressive

File: src/tests/networking_tests.rs [hunk 11]
```diff
@@ -65,159 +73,324 @@ pub mod net_tests {
...
+                assert_eq!(
+                    cage2.recvfrom_syscall(
+                        sockfd,
+                        buf.as_mut_ptr(),
+                        40,
+                        MSG_PEEK,
+                        &mut Some(&mut socket2)
+                    ),
+                    40
+                );
```
Comment (JustinCappos) [inline src/tests/networking_tests.rs:220]: too aggressive (like many others in this file)

**Review by JustinCappos — COMMENTED** [review]
So overall, I think this is helpful.  It is maybe a little overly aggressive at breaking up function arguments into separate lines (see my individual comments for some examples).  Also a few files need to be exempted.  However, on the whole this seems like it will be a nice cleanup.

Perhaps @rennergade and others should look at this also.  We likely should agree to a common rustfmt setting for the project and just exempt a few things here and there as is needed.

File: benches/fs_open_close.rs (hunk 1 above)
Comment (JustinCappos) [reply benches/fs_open_close.rs:57]: I've [read more about this](https://doc.rust-lang.org/nightly/style-guide/#trailing-commas) and withdraw this point.  It's weird to me, but I guess it is Rustic...

**Comment by JustinCappos** [comment]
Okay, I've looked a bit through the [Rust Style Guide](https://doc.rust-lang.org/nightly/style-guide).  I think I mostly agree with it, or at least, I trust the experts to know better than I would.  

I will also say though, that a lot of the fiddly bits I mention above are [small items](https://doc.rust-lang.org/nightly/style-guide/#small-items).  Or at least, seem like small items to me, which means they really should be inline instead of spread across 6-8 lines, as happens in some cases.  I don't know if it's worth changing rustfmt settings to handle this, so I'm okay letting that pass.

However, I do feel like Lind's constants files need to be exempted so their comments and "=" values line up in a more readable way.

I propose adding the following lines to `.rustfmt.toml`:

```ignore = [
    "src/safeposix/syscalls/net_constants.rs",
    "src/safeposix/syscalls/sys_constants.rs",
    "src/safeposix/syscalls/fs_constants.rs",
]```

**Comment by naveensrinivasan** [comment]
I didn't want to rebased and handle conflicts https://github.com/Lind-Project/safeposix-rust/pull/240 so I decided to close this.

**Pull request closed without merging**